app/models/test_setting.rb

```diff
@@ -1,17 +1,23 @@
 class TestSetting < ActiveRecord::Base
+  has_many :users, through: :tests
   has_many :tests
   has_many :questions
   belongs_to :subject
   belongs_to :categgory
   belongs_to :theme
+  belongs_to :user
+
+  class_attribute :current_user
 
   TEACHER_TESTSETTING_INDEX_PAGES = 10
+  TEACHER_TESTSETTING_SHOW_PAGES = 20
 
   resourcify#rolify
 
   accepts_nested_attributes_for :questions, allow_destroy: true
 
   scope :with_theme, ->(theme_id){ where(test_settings: {theme_id: theme_id} ) }
+  scope :current_teacher, ->{ where(user_id: current_user.id) }
```

Comment:
you can put params inside scope like 
```ruby
scope :current_teacher, -> (user_id){ where(user_id: user_id) }
```
and use it inside controller
```ruby
TestSetting.current_teacher(current_user.id)
```

Probably also rename scope `current_teacher` to `for_current_user` or to some similar, because it not clear what have you wanted to get.